src/expressions/postfix/evaluateLookup.ts

```diff
@@ -0,0 +1,75 @@
+import ArrayValue from '../dataTypes/ArrayValue';
+import ISequence from '../dataTypes/ISequence';
+import isSubtypeOf from '../dataTypes/isSubtypeOf';
+import MapValue from '../dataTypes/MapValue';
+import EmptySequence from '../dataTypes/Sequences/EmptySequence';
+import DynamicContext from '../DynamicContext';
+import ExecutionParameters from '../ExecutionParameters';
+import Expression from '../Expression';
+import isSameMapKey from '../functions/isSameMapKey';
+import concatSequences from '../util/concatSequences';
+import createDoublyIterableSequence from '../util/createDoublyIterableSequence';
+import Value from '../dataTypes/Value';
+
+function performLookup(
+	contextItem: Value,
+	lookup: '*' | Value,
+	previousSequence: ISequence
+): ISequence {
+	const sequences = [previousSequence];
+
+	if (isSubtypeOf(contextItem.type, 'array(*)')) {
+		const arrayItem = contextItem as ArrayValue;
+		if (lookup === '*') {
+			sequences.push(...arrayItem.members.map(member => member()));
+		} else if (!isSubtypeOf(lookup.type, 'xs:integer')) {
+			throw new Error('XPTY0004: The key specifier is not an integer.');
+		} else {
+			const index = lookup.value as number;
+			if (arrayItem.members.length < index || 0 >= index) {
+				throw new Error('FOAY0001: No item at specified index.');
```

Comment:
Recently, we're moving these kinds of errors to a centralized file. Could you consider that?